eclair-core/src/main/scala/fr/acinq/eclair/payment/send/PaymentInitiator.scala

```diff
@@ -51,19 +51,19 @@ class PaymentInitiator(nodeParams: NodeParams, outgoingPaymentFactory: PaymentIn
         // Immediately return the paymentId
         sender() ! paymentId
       }
-      val paymentCfg = SendPaymentConfig(paymentId, paymentId, r.externalId, r.paymentHash, r.recipientAmount, r.recipientNodeId, Upstream.Local(paymentId), Some(r.paymentRequest), storeInDb = true, publishEvent = true, recordPathFindingMetrics = true, Nil)
+      val paymentCfg = SendPaymentConfig(paymentId, paymentId, r.externalId, r.paymentHash, r.recipientAmount, r.recipientNodeId, Upstream.Local(paymentId), Some(r.invoice), storeInDb = true, publishEvent = true, recordPathFindingMetrics = true, Nil)
       val finalExpiry = r.finalExpiry(nodeParams.currentBlockHeight)
-      r.paymentRequest.paymentSecret match {
-        case _ if !r.paymentRequest.features.areSupported(nodeParams) =>
-          sender() ! PaymentFailed(paymentId, r.paymentHash, LocalFailure(r.recipientAmount, Nil, UnsupportedFeatures(r.paymentRequest.features.features)) :: Nil)
+      r.invoice.paymentSecret match {
+        case _ if !nodeParams.features.invoiceFeatures().areSupported(r.invoice.features) =>
```

Comment:
This has a side-effect, we will now be stricter than before and will refuse to pay invoices that set mandatory non-invoice bits (even if we support them in our `nodeParams.features`.

Have you checked whether other implementations correctly only include invoice feature bits in their invoice? Otherwise we may want to stay a bit more lenient here and do:

```suggestion
        case _ if !nodeParams.features.areSupported(r.invoice.features) =>
```

Reply:
This was actually part of typing the features, I missed it when I untyped them back.